### PR TITLE
Proxy app permissions

### PR DESCRIPTION
**Pull request by sdc50**
Add "Can Access App" permission for Proxy Apps

Add Users dialog to Change Group form

prevent apps from loading when migrating the database

### CODE REVIEW

**Comment by sdc50** [comment]
@swainn I'm not sure where to update images for the docs, but figure 4 on the admin_pages should be updated:

tethys_portal_assign_perm4

**Review by swainn — CHANGES_REQUESTED** [review]
Mainly suggestions for changes to the docs that you didn't write in the first place.

File: tethys_apps/apps.py [hunk 1]
```diff
@@ -20,6 +21,7 @@ def ready(self):
         """
         Startup method for Tethys Apps django app.
         """
-        # Perform App Harvesting
-        harvester = SingletonHarvester()
-        harvester.harvest()
+        if sys.argv[1] != "migrate":
+            # Perform App Harvesting (if database is not being migrated)
+            harvester = SingletonHarvester()
+            harvester.harvest()
```
[the remark this answers is not in the document]
Comment (sdc50) [reply tethys_apps/apps.py:27]: No, this is actually preventing an error from happening when migrating the DB for the first time. It tries to harvest the apps which access the database and then throws a bunch of errors saying the tables aren't there. The harvester handles them and doesn't load any apps, but this prevents all of the stack trace from being printed.

<details>

```python
Running migrations for Tethys database...
Traceback (most recent call last):
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
sqlite3.OperationalError: no such table: tethys_apps_tethysapp

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/rditlsc9/Workspace/tethys/tethys-dev/tethys_portal/manage.py", line 20, in <module>
    execute_from_command_line(sys.argv)
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/core/management/__init__.py", line 395, in execute
    django.setup()
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/contrib/admin/apps.py", line 27, in ready
    self.module.autodiscover()
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/contrib/admin/__init__.py", line 24, in autodiscover
    autodiscover_modules('admin', register_to=site)
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/utils/module_loading.py", line 47, in autodiscover_modules
    import_module('%s.%s' % (app_config.name, module_to_search))
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1206, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1178, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1149, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/Users/rditlsc9/Workspace/tethys/tethys-dev/tethys_apps/admin.py", line 190, in <module>
    class GOPAppAccessForm(forms.ModelForm):
  File "/Users/rditlsc9/Workspace/tethys/tethys-dev/tethys_apps/admin.py", line 218, in GOPAppAccessForm
    attrs={"data-app-packages": json.dumps({a.name: a.package for a in TethysApp.objects.all()})}
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/models/query.py", line 280, in __iter__
    self._fetch_all()
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/models/query.py", line 1324, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/models/query.py", line 51, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1175, in execute_sql
    cursor.execute(sql, params)
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in _execute
    with self.db.wrap_database_errors:
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rditlsc9/conda/envs/tethys-dev/lib/python3.11/site-packages/django/db/backends/sqlite3/base.py", line 423, in execute
    return Database.Cursor.execute(self, query, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.OperationalError: no such table: tethys_apps_tethysapp

Tethys Apps Loaded: 

WARNING:tethys.tethys_quotas.apps:Unable to sync resource quota handlers: No database found
WARNING:tethys.tethys_quotas.apps:Unable to sync resource quota handlers: No database found
Operations to perform:
  Apply all migrations: admin, auth, authtoken, axes, captcha, contenttypes, guardian, mfa, sessions, social_django, termsandconditions, tethys_apps, tethys_compute, tethys_config, tethys_quotas, tethys_services
Running migrations:
  Applying contenttypes.0001_initial... OK
  Applying auth.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_logentry_remove_auto_add... OK
  Applying admin.0003_logentry_add_action_flag_choices... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0002_alter_permission_name_max_length... OK
  Applying auth.0003_alter_user_email_max_length... OK
  Applying auth.0004_alter_user_username_opts... OK
  Applying auth.0005_alter_user_last_login_null... OK
  Applying auth.0006_require_contenttypes_0002... OK
  Applying auth.0007_alter_validators_add_error_messages... OK
  Applying auth.0008_alter_user_username_max_length... OK
  Applying auth.0009_alter_user_last_name_max_length... OK
  Applying auth.0010_alter_group_name_max_length... OK
  Applying auth.0011_update_proxy_permissions... OK
  Applying auth.0012_alter_user_first_name_max_length... OK
  Applying authtoken.0001_initial... OK
  Applying authtoken.0002_auto_20160226_1747... OK
  Applying authtoken.0003_tokenproxy... OK
  Applying axes.0001_initial... OK
  Applying axes.0002_auto_20151217_2044... OK
  Applying axes.0003_auto_20160322_0929... OK
  Applying axes.0004_auto_20181024_1538... OK
  Applying axes.0005_remove_accessattempt_trusted... OK
  Applying axes.0006_remove_accesslog_trusted... OK
  Applying axes.0007_alter_accessattempt_unique_together... OK
  Applying axes.0008_accessfailurelog... OK
  Applying captcha.0001_initial... OK
  Applying captcha.0002_alter_captchastore_id... OK
  Applying guardian.0001_initial... OK
  Applying guardian.0002_generic_permissions_index... OK
  Applying mfa.0001_initial... OK
  Applying mfa.0002_user_keys_key_type... OK
  Applying mfa.0003_auto_20181114_2159... OK
  Applying mfa.0004_user_keys_enabled... OK
  Applying mfa.0005_auto_20181115_2014... OK
  Applying mfa.0006_trusted_devices... OK
  Applying mfa.0007_auto_20181230_1549... OK
  Applying mfa.0008_user_keys_last_used... OK
  Applying mfa.0009_user_keys_owned_by_enterprise... OK
  Applying mfa.0010_auto_20201110_0557... OK
  Applying mfa.0011_auto_20210530_0622... OK
  Applying sessions.0001_initial... OK
  Applying social_django.0001_initial... OK
  Applying social_django.0002_add_related_name... OK
  Applying social_django.0003_alter_email_max_length... OK
  Applying social_django.0004_auto_20160423_0400... OK
  Applying social_django.0005_auto_20160727_2333... OK
  Applying social_django.0006_partial... OK
  Applying social_django.0007_code_timestamp... OK
  Applying social_django.0008_partial_timestamp... OK
  Applying social_django.0009_auto_20191118_0520... OK
  Applying social_django.0010_uid_db_index... OK
  Applying social_django.0011_alter_id_fields... OK
  Applying termsandconditions.0001_initial... OK
  Applying termsandconditions.0002_termsandconditions_info... OK
  Applying termsandconditions.0003_auto_20170627_1217... OK
  Applying termsandconditions.0004_auto_20201107_0711... OK
  Applying tethys_services.0001_initial_40... OK
  Applying tethys_compute.0001_initial_41... OK
  Applying tethys_apps.0001_initial_41... OK
  Applying tethys_config.0001_initial_40... OK
  Applying tethys_quotas.0001_initial_40... OK
Creating Tethys Portal superuser "admin"...

```

</details>

File: docs/tethys_portal/admin_pages.rst [hunk 2]
```diff
@@ -68,12 +68,12 @@ Administrators of Tethys Portal can control access to each app in the portal. Ho
...
+To assign app permissions to a user using the ``Change Group`` method, go to the administrator dashboard and select the ``Groups`` link under the ``Authentication and Authorization`` heading. Select the link with the group name from the list. On the ``Change Group`` page you can assign a group permissions to access a specific proxy app from the ``Proxy Apps`` multiselect field or installed app from the ``Apps`` multiselect field by moving an app from the available app box to the chosen app box (see Figure 4). When an installed app is added to the chosen apps box, the form will dynamically display all the permissions and groups associated with that app if there are any (see Figure 5). Individual permissions can be added to the group by moving the permissions from the available box of the specific app to the chosen box. In addition, all the permissions from another group that are associated to the specific app can also be added by moving the specific group from the available groups box to the chosen groups box. The ``Change Group`` form then needs to be saved for changes to take effect.
 
 .. figure:: ../images/tethys_portal/tethys_portal_assign_perm4.png
```
[the remark this answers is not in the document]
Comment (sdc50) [reply docs/tethys_portal/admin_pages.rst:96]: It looks like `git lfs` is smarter than I realized. I think I just replace the file and then `lfs` intercepts it and does what it needs to.

File: tethys_apps/apps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdc50) [reply tethys_apps/apps.py:27]: @swainn if you're okay with this change then this PR should be ready (if stickler ever decides to run).

**Pull request merged**